Source/WebKit/NetworkProcess/webrtc/NetworkRTCMonitor.cpp

```diff
@@ -266,8 +266,11 @@ static std::optional<RTCNetwork::IPAddress> getSocketLocalAddress(int socket, bo
 {
     sockaddr_storage localAddressStorage;
     zeroBytes(localAddressStorage);
-    socklen_t localAddressStorageLength = sizeof(sockaddr_storage);
-    if (::getsockname(socket, reinterpret_cast<sockaddr*>(&localAddressStorage), &localAddressStorageLength) < 0) {
+
+    auto& localAddress = asSocketAddress(localAddressStorage);
```

Comment:
Since we haven’t initialized the bytes yet, we have no type yet, so I’m surprised that this type safe cast doesn’t crash. Are we sure this is right?

Reply:
That's a good point. The code was previously just doing a `reinterpret_cast` without initializing the bytes too. In order to initialize the bytes as either IPv4 or 6 we'd need a public host and port it seems (the way `connectToRemoteAddress` in that same file does). Do we know if it's ok to use those same values?

Reply:
> Since we haven’t initialized the bytes yet, we have no type yet, so I’m surprised that this type safe cast doesn’t crash. Are we sure this is right?

asSocketAddress() is safe because it takes a `sockaddr_storage` in and a `sockaddr_storage` is wide enough to store both IPv4 and IPv6. All it does is thus a reinterpret_cast, no type check, by design. We're still building the address.